### PR TITLE
[eclipse] new workflow to publish web app on linux 

### PR DESCRIPTION
**Pull request by Eskibear**
image

image

### CODE REVIEW

File: PluginsAndFeatures/azure-toolkit-for-eclipse/com.microsoft.azuretools.container/src/com/microsoft/azuretools/container/utils/DockerUtil.java [hunk 1]
```diff
@@ -87,10 +87,10 @@ public static Container runContainer(@NotNull DockerClient docker, @NotNull Stri
     /**
      * buildImage.
      */
-    public static String buildImage(@NotNull DockerClient docker, @NotNull String imageNameWithTag,
-            @NotNull Path dockerDirectory, ProgressHandler progressHandler)
+    public static String buildImage(DockerClient docker, String imageNameWithTag, Path dockerDirectory,
```
Comment (adashen) [inline PluginsAndFeatures/azure-toolkit-for-eclipse/com.microsoft.azuretools.container/src/com/microsoft/azuretools/container/utils/DockerUtil.java]: Add NotNull

File: Utils/azure-explorer-common/src/com/microsoft/tooling/msservices/serviceexplorer/azure/container/WebAppOnLinuxDeployPresenter.java [hunk 2]
```diff
@@ -152,7 +153,9 @@ public void onLoadPricingTierList() {
      */
     public void onLoadAppServicePlan(String sid, String rg) {
         Observable.fromCallable(() -> AzureWebAppMvpModel.getInstance()
-                .listAppServicePlanBySubscriptionIdAndResourceGroupName(sid, rg))
+                .listAppServicePlanBySubscriptionIdAndResourceGroupName(sid, rg).stream()
```
Comment (adashen) [inline Utils/azure-explorer-common/src/com/microsoft/tooling/msservices/serviceexplorer/azure/container/WebAppOnLinuxDeployPresenter.java]: also remove the filter in view of intelliJ

File: PluginsAndFeatures/azure-toolkit-for-eclipse/com.microsoft.azuretools.container/src/com/microsoft/azuretools/container/ui/WebAppTableComposite.java [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+import org.eclipse.swt.widgets.Composite;
+import org.eclipse.swt.widgets.Table;
+import org.eclipse.swt.SWT;
+import org.eclipse.swt.layout.FillLayout;
+import org.eclipse.swt.widgets.TableColumn;
+import org.eclipse.swt.widgets.Button;
+import org.eclipse.swt.layout.GridLayout;
+import org.eclipse.swt.layout.GridData;
+
+public class WebAppTableComposite extends Composite {
```
Comment (adashen) [inline PluginsAndFeatures/azure-toolkit-for-eclipse/com.microsoft.azuretools.container/src/com/microsoft/azuretools/container/ui/WebAppTableComposite.java]: remove public

File: PluginsAndFeatures/azure-toolkit-for-eclipse/com.microsoft.azuretools.container/src/com/microsoft/azuretools/container/ui/NewWebAppComposite.java [hunk 4]
```diff
@@ -0,0 +1,152 @@
...
+import org.eclipse.swt.widgets.ExpandBar;
+import org.eclipse.swt.widgets.ExpandItem;
+import org.eclipse.swt.layout.GridLayout;
+import org.eclipse.swt.layout.GridData;
+import org.eclipse.swt.widgets.Label;
+import org.eclipse.swt.widgets.Text;
+import org.eclipse.swt.widgets.Combo;
+import org.eclipse.swt.widgets.Button;
+
+public class NewWebAppComposite extends Composite {
```
Comment (adashen) [inline PluginsAndFeatures/azure-toolkit-for-eclipse/com.microsoft.azuretools.container/src/com/microsoft/azuretools/container/ui/NewWebAppComposite.java]: remove public

File: PluginsAndFeatures/azure-toolkit-for-eclipse/com.microsoft.azuretools.container/src/com/microsoft/azuretools/container/ui/PublishWebAppOnLinuxDialog.java [hunk 5]
```diff
@@ -0,0 +1,726 @@
...
+import com.microsoft.azuretools.utils.AzureUIRefreshCore;
+import com.microsoft.azuretools.utils.AzureUIRefreshEvent;
+import com.microsoft.tooling.msservices.serviceexplorer.azure.container.WebAppOnLinuxDeployPresenter;
+import com.microsoft.tooling.msservices.serviceexplorer.azure.container.WebAppOnLinuxDeployView;
+import com.spotify.docker.client.DefaultDockerClient;
+import com.spotify.docker.client.DockerClient;
+
+import rx.Observable;
+
+public class PublishWebAppOnLinuxDialog extends TitleAreaDialog implements WebAppOnLinuxDeployView {
```
Comment (adashen) [inline PluginsAndFeatures/azure-toolkit-for-eclipse/com.microsoft.azuretools.container/src/com/microsoft/azuretools/container/ui/PublishWebAppOnLinuxDialog.java]: extend from azure wrapper

File: PluginsAndFeatures/azure-toolkit-for-eclipse/com.microsoft.azuretools.container/src/com/microsoft/azuretools/container/ui/PublishWebAppOnLinuxDialog.java [hunk 6]
```diff
@@ -0,0 +1,726 @@
...
+        if (ret) {
+            webAppOnLinuxDeployPresenter.onDetachView();
+        }
+        return ret;
+    }
+
+    @Override
+    protected void okPressed() {
+        boolean validated = false;
+        loadDataModel();
```
Comment (adashen) [inline PluginsAndFeatures/azure-toolkit-for-eclipse/com.microsoft.azuretools.container/src/com/microsoft/azuretools/container/ui/PublishWebAppOnLinuxDialog.java]: rename to apply() ?

File: PluginsAndFeatures/azure-toolkit-for-eclipse/com.microsoft.azuretools.container/src/com/microsoft/azuretools/container/ui/PublishWebAppOnLinuxDialog.java [hunk 7]
```diff
@@ -0,0 +1,726 @@
...
+        try {
+            validation();
+            validated = true;
```
Comment (adashen) [inline PluginsAndFeatures/azure-toolkit-for-eclipse/com.microsoft.azuretools.container/src/com/microsoft/azuretools/container/ui/PublishWebAppOnLinuxDialog.java]: move execute to this try block

File: PluginsAndFeatures/azure-toolkit-for-eclipse/com.microsoft.azuretools.container/src/com/microsoft/azuretools/container/ui/PublishWebAppOnLinuxDialog.java [hunk 8]
```diff
@@ -0,0 +1,726 @@
...
+        acrHolder.setExpanded(true);
+        acrHolder.setText("Azure Container Registry");
+
+        cpAcr = new Composite(expandBar, SWT.NONE);
+        acrHolder.setControl(cpAcr);
+        acrHolder.setHeight(acrHolder.getControl().computeSize(SWT.DEFAULT, SWT.DEFAULT).y);
+
+        webappHolder = new ExpandItem(expandBar, SWT.NONE);
+        webappHolder.setExpanded(true);
+        webappHolder.setText("Web App On Linux");
```
Comment (adashen) [inline PluginsAndFeatures/azure-toolkit-for-eclipse/com.microsoft.azuretools.container/src/com/microsoft/azuretools/container/ui/PublishWebAppOnLinuxDialog.java]: string to constant? And the name should be "Web App for Container" ?

File: PluginsAndFeatures/azure-toolkit-for-eclipse/com.microsoft.azuretools.container/src/com/microsoft/azuretools/container/ui/PublishWebAppOnLinuxDialog.java [hunk 9]
```diff
@@ -0,0 +1,726 @@
...
+            throw new InvalidDataException(MISSING_ARTIFACT);
+        }
+        if (!model.getTargetName().matches(ARTIFACT_NAME_REGEX)) {
+            throw new InvalidDataException(String.format(INVALID_WAR_FILE, model.getTargetName()));
+        }
+    }
+
+    private void execute() {
+        Observable.fromCallable(() -> {
+            ConsoleLogger.info("Starting job ...  ");
```
Comment (adashen) [inline PluginsAndFeatures/azure-toolkit-for-eclipse/com.microsoft.azuretools.container/src/com/microsoft/azuretools/container/ui/PublishWebAppOnLinuxDialog.java]: check the existing processHandler

File: Utils/azure-explorer-common/src/com/microsoft/tooling/msservices/serviceexplorer/azure/container/WebAppOnLinuxDeployPresenter.java (hunk 2 above)
Comment (Eskibear) [reply Utils/azure-explorer-common/src/com/microsoft/tooling/msservices/serviceexplorer/azure/container/WebAppOnLinuxDeployPresenter.java]: agree

File: PluginsAndFeatures/azure-toolkit-for-eclipse/com.microsoft.azuretools.container/src/com/microsoft/azuretools/container/ui/WebAppTableComposite.java [hunk 10]
```diff
@@ -0,0 +1,83 @@
...
+        super(parent, style);
+        setLayout(new FillLayout(SWT.HORIZONTAL));
+
+        Composite composite_1 = new Composite(this, SWT.NONE);
+        composite_1.setLayout(new GridLayout(2, false));
+
+        tblWebApps = new Table(composite_1, SWT.BORDER | SWT.FULL_SELECTION);
+        GridData gd_tblWebApps = new GridData(SWT.FILL, SWT.FILL, true, true, 1, 1);
+        gd_tblWebApps.minimumHeight = 200;
+        gd_tblWebApps.widthHint = 331;
```
Comment (jdneo) [inline PluginsAndFeatures/azure-toolkit-for-eclipse/com.microsoft.azuretools.container/src/com/microsoft/azuretools/container/ui/WebAppTableComposite.java]: Can we let the table fill the space?

File: PluginsAndFeatures/azure-toolkit-for-eclipse/com.microsoft.azuretools.container/src/com/microsoft/azuretools/container/ui/NewWebAppComposite.java [hunk 11]
```diff
@@ -0,0 +1,174 @@
...
+
+    /**
+     * Create the composite.
+     *
+     * @param parent
+     * @param style
+     */
+    public NewWebAppComposite(Composite parent, int style) {
+        super(parent, style);
+        setLayout(new FillLayout(SWT.HORIZONTAL));
```
Comment (jdneo) [inline PluginsAndFeatures/azure-toolkit-for-eclipse/com.microsoft.azuretools.container/src/com/microsoft/azuretools/container/ui/NewWebAppComposite.java]: root layout need to be a fill layout here?

File: PluginsAndFeatures/azure-toolkit-for-eclipse/com.microsoft.azuretools.container/src/com/microsoft/azuretools/container/ui/NewWebAppComposite.java [hunk 12]
```diff
@@ -0,0 +1,174 @@
...
+import org.eclipse.swt.widgets.Button;
+import org.eclipse.swt.widgets.Combo;
+import org.eclipse.swt.widgets.Composite;
+import org.eclipse.swt.widgets.ExpandBar;
+import org.eclipse.swt.widgets.ExpandItem;
+import org.eclipse.swt.widgets.Label;
+import org.eclipse.swt.widgets.Text;
+
+class NewWebAppComposite extends Composite {
+    Text txtAppName;
```
Comment (jdneo) [inline PluginsAndFeatures/azure-toolkit-for-eclipse/com.microsoft.azuretools.container/src/com/microsoft/azuretools/container/ui/NewWebAppComposite.java]: will be better if add ```private``` decorator

File: PluginsAndFeatures/azure-toolkit-for-eclipse/com.microsoft.azuretools.container/src/com/microsoft/azuretools/container/ui/NewWebAppComposite.java [hunk 13]
```diff
@@ -0,0 +1,174 @@
...
+     * Create the composite.
+     *
+     * @param parent
+     * @param style
+     */
+    public NewWebAppComposite(Composite parent, int style) {
+        super(parent, style);
+        setLayout(new FillLayout(SWT.HORIZONTAL));
+
+        Composite composite_2 = new Composite(this, SWT.NONE);
```
Comment (jdneo) [inline PluginsAndFeatures/azure-toolkit-for-eclipse/com.microsoft.azuretools.container/src/com/microsoft/azuretools/container/ui/NewWebAppComposite.java]: will be better if give them more beautiful variable name

File: PluginsAndFeatures/azure-toolkit-for-eclipse/com.microsoft.azuretools.container/src/com/microsoft/azuretools/container/ui/PublishWebAppOnLinuxDialog.java [hunk 14]
```diff
@@ -173,7 +169,7 @@ protected Control createDialogArea(Composite parent) {
 
         webappHolder = new ExpandItem(expandBar, SWT.NONE);
         webappHolder.setExpanded(true);
-        webappHolder.setText("Web App On Linux");
+        webappHolder.setText("Web App for Container");
```
Comment (adashen) [inline PluginsAndFeatures/azure-toolkit-for-eclipse/com.microsoft.azuretools.container/src/com/microsoft/azuretools/container/ui/PublishWebAppOnLinuxDialog.java]: should be "Web App for Containers" And need to move to constant

**Pull request merged**